storage/block/result.go

```diff
@@ -151,9 +156,12 @@ func (s *fetchBlocksMetadataResults) Reset() {
 }
 
 func (s *fetchBlocksMetadataResults) Close() {
-	for _, res := range s.results {
-		res.Blocks.Close()
+	var zeroed FetchBlocksMetadataResult
+	for i := range s.results {
+		s.results[i].Blocks.Close()
+		s.results[i] = zeroed
 	}
+	s.results = s.results[:0]
```

Comment:
Same here, should do it inside `fetchBlocksMetadataResultsPool.Put()` for reasons mentioned above